fix: handle panel ID correctly in make:filament-resource command

mwguerra opened the pull request:
**Description**:
This pull request addresses issue #13081. The `make:filament-resource` command was not handling panel IDs correctly when the `--panel` flag was used. The command now correctly checks the panel ID, attempts to match using `Str::lcfirst`, and provides an error message if the panel is not found.

**Visual changes**:
There are no visual changes associated with this PR.

**Functional changes**:
- [x] Code style has been fixed by running the `composer cs` command.
- [x] Changes have been tested to not break existing functionality.
- [x] Documentation is up-to-date.

Closes #13081

**Code Changes**:
Here are the relevant changes in the MakeResourceCommand.php:
```php
<?php

namespace Filament\Commands;

use Filament\Clusters\Cluster;
use Filament\Facades\Filament;
use Filament\Forms\Commands\Concerns\CanGenerateForms;
use Filament\Panel;
use Filament\Support\Commands\Concerns\CanIndentStrings;
use Filament\Support\Commands\Concerns\CanManipulateFiles;
use Filament\Support\Commands\Concerns\CanReadModelSchemas;
use Filament\Tables\Commands\Concerns\CanGenerateTables;
use Illuminate\Console\Command;
use Illuminate\Support\Arr;
use Illuminate\Support\Str;
use Symfony\Component\Console\Attribute\AsCommand;

use function Laravel\Prompts\select;
use function Laravel\Prompts\text;

#[AsCommand(name: 'make:filament-resource')]
class MakeResourceCommand extends Command
{
    use CanGenerateForms;
    use CanGenerateTables;
    use CanIndentStrings;
    use CanManipulateFiles;
    use CanReadModelSchemas;

    protected $description = 'Create a new Filament resource class and default page classes';

    protected $signature = 'make:filament-resource {name?} {--model-namespace=} {--soft-deletes} {--view} {--G|generate} {--S|simple} {--panel=} {--model} {--migration} {--factory} {--F|force}';

    public function handle(): int
    {
        // Existing code...

        $panelOption = Str::lcfirst($this->option('panel'));

        if ($panelOption) {
            $panel = Filament::getPanel($panelOption);

            if ($panel->getId() !== $panelOption) {
                $this->components->error("Panel [{$panelOption}] not found.");
                return static::INVALID;
            }
        }

        if (! $panelOption) {
            $panels = Filament::getPanels();

            /** @var Panel $panel */
            $panel = (count($panels) > 1) ? $panels[select(
                label: 'Which panel would you like to create this in?',
                options: array_map(
                    fn (Panel $panel): string => $panel->getId(),
                    $panels,
                ),
                default: Filament::getDefaultPanel()->getId()
            )] : Arr::first($panels);
        }

        // Rest of the code...
    }
}
```

mwguerra committed:
fix: handle panel ID correctly in make:filament-resource command

danharrin reviewed (changes requested):
I think the fix belongs in `PanelRegistry::get()`. If an ID is not null but not found, it should throw an exception. If the ID is not passed, it should return the default.

Instead of `$this->panels[$id]`, you should create a temporary array where all the panel IDs are fully lowercase and have `-` and `_` removed. Then do the same to the ID. Then the method can match `FooBar` to `foobar` `fooBar` `FoObAr`, `foo-bar`, `foo_bar` as well

danharrin commented:
Actually, I think there should be a new `$isStrict = true` method on the `get()` of the Filament facade, manager, and panel registry, which optionally uses this new index of panels. As we don't need that behaviour in every place.

Then we pass through `isStrict: false` from all the commands. It's not just the resource one, theres others like `make:filament-page`, make sure to check all usages

mwguerra committed:
Revert "fix: handle panel ID correctly in make:filament-resource command"

This reverts commit ed5277f5f21ce1b94e8ef0558d68e80dfab828f2.

mwguerra committed:
refactor: improve getPanel method with isStrict parameter and fallback

- Introduced default isStrict parameter set to true for getPanel method.
- Extracted panel retrieval logic into protected findPanel method.
- findPanel method strictly adheres to isStrict parameter, returning null if panel is not found.
- Maintained fallback in getPanel method to return default panel if not found.
- Ensured commands with direct user input set isStrict to false.
- This approach balances strictness with user experience, preventing potential issues in methods like getRouteName.

mwguerra commented:
I've made some updates to the `getPanel` method and introduced a default `isStrict` parameter set to `true`. This means that, by default, the method operates in strict mode unless specified otherwise. Only when commands directly involving user input are executed is `isStrict` set to `false`.

For better separation of concerns and improved readability, I extracted the logic into a protected `findPanel` method. This method adheres to the `isStrict` parameter and returns `null` if a panel isn't found, ensuring strict behavior as intended.

However, the `getPanel` method still maintains the original fallback mechanism. If no panel is found, it returns the default panel. This approach ensures that we avoid user experience issues that could arise from strictly returning `null`. For instance, methods like `getRouteName` could be negatively impacted if they attempt to access a non-existent panel.

If you have any questions or need further adjustments, I'm happy to help. Just let me know!

danharrin committed:
Update PanelRegistry.php

danharrin commented:
Perfect, thanks for understanding!
